core/services/relay/evm/evm.go

```diff
@@ -75,18 +76,39 @@ func (r *Relayer) NewConfigProvider(args relaytypes.RelayArgs) (relaytypes.Confi
 
 type configWatcher struct {
 	utils.StartStopOnce
+	lggr             logger.Logger
 	contractAddress  common.Address
 	contractABI      abi.ABI
 	offchainDigester types.OffchainConfigDigester
 	configPoller     *ConfigPoller
 	chain            evm.Chain
+	new              bool
+	fromBlock        uint64
+	replayCtx        context.Context
+	replayCancel     context.CancelFunc
+	wg               sync.WaitGroup
 }
 
 func (c *configWatcher) Start(ctx context.Context) error {
+	if c.new {
+		// Only replay if its a brand new job.
+		c.wg.Add(1)
+		go func() {
+			defer c.wg.Done()
+			c.lggr.Infow("starting replay for config", "fromBlock", c.fromBlock)
+			if err := c.configPoller.destChainLogPoller.Replay(context.Background(), int64(c.fromBlock)); err != nil {
```

Comment:
Is the `replayCtx` meant to go here?
```suggestion
			if err := c.configPoller.destChainLogPoller.Replay(replayCtx, int64(c.fromBlock)); err != nil {
```